### DISCUSSION TITLE
One liner to construct mixed color text for console.print?

### DISCUSSION THREAD

**Discussion by hemna** [Textualize/rich#1775] [Q&A] [answered]
Is there an easy way to construct a rich text string that has multiple colored components on a single line?

For example, I want to construct a string like

f"This is a {test} and another {test2} and something"

Where test is green and test2 is blue.

From what I can tell, in order to do this I'd have to do something like:

```
test = 'something'
test2 = 'somethingelse'  

msg = Text("This is a ")
msg.append(test, style="green")
msg.append(" and another ")
msg.append(test2, style="blue")
msg.append(" and something")

console.print(msg)
```
Is there a 1 liner that I can accomplish the same thing?

**Comment by willmcgugan** [comment, marked as answer]
Two options:

```python
msg = Text.from_markup("This is a [green]test[/] and another [blue]test2[/] and something")
```

```python
msg = Text.assemble(
    "This is a ",
    ("test", "green"),
    " and another ",
    ("test2, blue"),
    " and something"
)
```

**Reply by hemna** [reply]
Awesome thank you.
